lib/pact_broker/client/webhooks/create.rb

```diff
@@ -36,7 +36,11 @@ def call
         attr_reader :http_client
 
         def create_webhook_with_consumer_and_provider
-          webhook_entity = webhook_link.expand(consumer: params.consumer, provider: params.provider).post(request_body)
+          if params.uuid
```

Comment:
`create_webhook_with_consumer_and_provider` just supports the original use case where the consumer and provider were both mandatory, and the URL had to contain the consumer and provider names. It's just kept here for backwards compatiblity.

The new way of creating webhooks is to put the consumer and provider into the body of the request, and do a POST to the `/webhooks` collection  - or, as you are now adding code for, a PUT to `/webhooks/{UUID}`

These new lines needs to go into the `create_webhook_with_optional_consumer_and_provider` method instead. You'll have to change the if statement in the `call` method to also use `create_webhook_with_optional_consumer_and_provider` if a uuid has been supplied.

Reply:
All the lines that have been added to this method need to be removed and put into `create_webhook_with_optional_consumer_and_provider`. This method is only for creating webhooks using the old URL format. It won't support the new URL format.

Reply:
I think some of what you've written applied when we were working with the existing `create-webhook` command. Now that we've got a new command `create-or-update-webhook`, I've moved all the logic into the `create_webhook_with_optional_consumer_and_provider` method, and if params are present, we make a PUT request with the `uuid`, otherwise, we make the standard POST request